packages/opentelemetry/lib/tracing/tracerProvider.js

```diff
@@ -0,0 +1,139 @@
+/* eslint-disable no-unused-vars */
+
+import { getBatchSpanProcessor } from './spanProcessor';
+import { buildExporter, AVAILABLE_EXPORTERS } from './exporter';
+import { NodeTracerProvider } from '@opentelemetry/node';
+import { registerInstrumentations, InstrumentationOption } from '@opentelemetry/instrumentation';
+import { ServerInstrumentation } from './serverInstrumenation';
+import { diag, DiagConsoleLogger, DiagLogLevel } from '@opentelemetry/api';
+import { SpanProcessor, } from '@opentelemetry/tracing';
+import { ExporterConfig } from '@opentelemetry/exporter-jaeger';
+import _ from 'lodash';
+
+//Delegate
+class TracerProvider {
+  constructor () {
+    this.provider = new NodeTracerProvider();
+    this.exporters = {};
+    this.init();
+  }
+
+  /**
+   * @typedef {Object} exporter
+   * @property {string} exporter_type - exporterType of the exporter
+   * @property {ExporterConfig} config - config for that specific exporter_type
+    */
+
+  init () {
+    diag.setLogger(new DiagConsoleLogger(), DiagLogLevel.ALL);
+    this.provider.register();
+
+    const consoleExporter = buildExporter(AVAILABLE_EXPORTERS.CONSOLE);
+    const spanProcessor = getBatchSpanProcessor(consoleExporter);
+    this.addSpanProcessor(spanProcessor);
+    this.exporters[AVAILABLE_EXPORTERS.CONSOLE] = consoleExporter;
+
+    this._serverInstrumentation = new ServerInstrumentation();
+    registerInstrumentations({
+      instrumentation: this._serverInstrumentation.instrumentations,
+      tracerProvider: this.provider
+    });
+    this._currentConfig = {
+      active: true,
+      currentExporters: [AVAILABLE_EXPORTERS.CONSOLE],
+      exporters:
+        [
+          {
+            exporter_type: 'console',
```

Comment:
could the key name be in camelCase or this is the lib requirement to have it like that?

Reply:
nope this is our definition, can be moved to camel case